tests/tt_eager/python_api_testing/unit_testing/misc/test_softmax_sharded.py

```diff
@@ -351,35 +319,26 @@ def test_softmax_with_sharded_mask(device, in_dtype, in0_mem_config, shard_orien
     attention_mask = torch.rand(1, 32, 32, 1024)
     attention_mask = (attention_mask > 0.5).float()
     attention_mask = torch.where(attention_mask == 1, torch.tensor(0.0), torch.tensor(-float("inf")))
-    attention_mask_t = torch2tt_tensor(
-        attention_mask,
-        device,
-        tt_memory_config=in0_mem_config,
-        tt_dtype=(
-            ttl.tensor.DataType.FLOAT32 if in_dtype == ttl.tensor.DataType.FLOAT32 else ttl.tensor.DataType.BFLOAT16
-        ),
-    )
-    attention_mask_t_shard = ttl.tensor.interleaved_to_sharded(
-        attention_mask_t,
-        grid_size,
-        [M, K],
-        ttl.tensor.TensorMemoryLayout.HEIGHT_SHARDED,
-        shard_orient,
+    mask_dtype = ttnn.float32 if in_dtype == ttnn.float32 else ttnn.bfloat16
+    attention_mask_t = ttnn.from_torch(
+        attention_mask, dtype=mask_dtype, layout=ttnn.TILE_LAYOUT, device=device, memory_config=in0_mem_config
     )
+    grid_coord = ttnn.CoreCoord(grid_size[0] - 1, grid_size[1] - 1)
+    shard_grid = ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), grid_coord)})
+    shard_shape = [M, K]
+    shard_spec = ttnn.ShardSpec(shard_grid, shard_shape, shard_orient, False)
+    sharded_mem_config = ttnn.MemoryConfig(ttnn.TensorMemoryLayout.HEIGHT_SHARDED, ttnn.BufferType.L1, shard_spec)
+    attention_mask_t_shard = ttnn.to_memory_config(attention_mask_t, sharded_mem_config)
```

Comment:
this code is repeated in so many places, may be a great idea to have a function for this